privacy-on-beam/pbeam/distinct_id_test.go

```diff
@@ -275,6 +319,38 @@ func TestDistinctPrivacyIDCrossPartitionContributionBounding(t *testing.T) {
 	}
 }
 
+// Checks that DistinctPrivacyID bounds cross-partition contributions correctly.
+// The logic mirrors TestCountCrossPartitionContributionBounding.
+func TestDistinctPrivacyIDWithPartitionsCrossPartitionContributionBounding(t *testing.T) {
+	// pairs contains {1,0}, {2,0}, …, {50,0}, {1,1}, …, {50,1}, {1,2}, …, {50,9}.
+	var pairs []pairII
+	for i := 0; i < 10; i++ {
+		pairs = append(pairs, makePairsWithFixedV(50, i)...)
+	}
+	result := []testInt64Metric{
+		{0, 150},
+	}
+	p, s, col, want := ptest.CreateList2(pairs, result)
+	col = beam.ParDo(s, pairToKV, col)
+	partitionsCol := CreateList(s, []int{0, 1, 2})
```

Comment:
We are not actually testing cross-partition contribution bounding if we have a total of 3 kept partitions. We should keep more partitions and let Privacy on Beam drop the extra ones. So, let's say you keep `{0, 1, 2, 3, 4}`.